### PR TITLE
BYO scaledobject

### LINKED ISSUE #13: Allow non-managed mode
- Users might want to create Keda scaled objects on their own. Right now by default we create the ScaledObject (managed mode). The autoscaler-keda expects the hpa object to be created/initialized correctly as a contract, and it will not interfere with KEDA in the non-managed mode (BYO scaledobject).
- Non-managed mode allows to support the full Keda spec. Since we are not embedding the keda spec in the ksvc and we want to keep things simple in the future (as KEDA evolves as well), we will not map everything from the KEDA spec via annotations in the managed mode.

### PR DESCRIPTION
**Pull request by skonto**
# Changes

- Allow user to define their own scaled object. This can be done per service or globally.

/kind <kind>

Fixes #13 

**Release Note**

```release-note
BYO scaledobject is now supported and can be configured per service.
```

### CODE REVIEW

File: test/e2e/autoscale_custom_test.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func setupCustomHPASvc(t *testing.T, metric string, target int) *TestContext {
 				autoscaling.MinScaleAnnotationKey:                "1",
 				autoscaling.MaxScaleAnnotationKey:                fmt.Sprintf("%d", int(maxPods)),
 				autoscaling.WindowAnnotationKey:                  "20s",
-				resources2.KedaAutoscaleAnotationPrometheusQuery: "sum(rate(http_requests_total{}[1m]))",
+				resources2.KedaAutoscaleAnotationPrometheusQuery: fmt.Sprintf("sum(rate(http_requests_total{namespace='%s'}[1m]))", test.ServingFlags.TestNamespace),
```
Comment (skonto) [inline test/e2e/autoscale_custom_test.go:85]: Should be specific to the namespace where metrics are emitted.

File: pkg/reconciler/autoscaling/hpa/keda_hpa.go [hunk 2]
```diff
@@ -42,7 +46,10 @@ import (
 	"knative.dev/autoscaler-keda/pkg/reconciler/autoscaling/hpa/resources"
 )
 
-const allActivators = 0
+const (
+	allActivators                    = 0
+	KedaAutoscaleAnotationAutocreate = autoscaling.GroupName + "/scaled-object-auto-create"
```
Comment (ReToCode) [inline pkg/reconciler/autoscaling/hpa/keda_hpa.go:51]: `KedaAutoscaleAnnotationAutocreate`

File: config/config.yaml [hunk 3]
```diff
@@ -40,3 +40,12 @@ data:
...
     # configures the default Prometheus address if no is specified per service
     autoscaler.keda.prometheus-address: "http://prometheus-operated.default.svc:9090"
+
+    # configures the globally default mode for creating ScaledObject. Default is true.
+    # If you set this to false make sure that you have a ScaledObject for each service
+    # and existing services with a previously automatically created ScaledObject should
+    # have their Knative service annotation `autoscaling.knative.dev/scaled-object-auto-create` set to true.
+    # By setting `autoscaling.knative.dev/scaled-object-auto-create` at the Knative Service level you can bypass
+    # this configuration and by setting to false you can bring your own scaled object.
+
```
Comment (ReToCode) [inline config/config.yaml:50]: probably no newline here?

**Review by ReToCode — COMMENTED** [review]
only nit stuff

File: pkg/reconciler/autoscaling/hpa/keda_hpa.go (hunk 2 above)
Comment (skonto) [reply pkg/reconciler/autoscaling/hpa/keda_hpa.go]: I see the same spelling error elsewhere. I am going to fix it.

**Comment by skonto** [comment]
@ReToCode I think this is ready.

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

**Pull request merged**